algorithm/cgal/arrangeSegments.js

```diff
@@ -10,46 +10,59 @@ export const arrangeSegments = (
   segments,
   triangulate = false
 ) => {
-  const c = getCgal();
-  let target;
-  let polygon;
-  let polygons = [];
-  const fill = (out) => {
-    for (const [start, end] of segments) {
-      c.addPoint(out, start[X], start[Y], start[Z]);
-      c.addPoint(out, end[X], end[Y], end[Z]);
-    }
-  };
-  const emitPolygon = (isHole) => {
-    target = { points: [], exactPoints: [], holes: [], plane, exactPlane };
-    if (isHole) {
-      polygon.holes.push(target);
+  try {
+    const c = getCgal();
```

Comment:
getCgal() returns a new instance, right? It's not a singleton?

Reply:
It returns a singleton, allocated on the first call.